bentoml/_internal/frameworks/transformers.py

```diff
@@ -106,427 +65,209 @@ def check_flax_supported() -> None:  # pragma: no cover
-            accordingly to your use case provided by :obj:`transformers`. Refers to
-            `Feature Extractor API <https://huggingface.co/docs/transformers/main_classes/feature_extractor>`_
-            for more information
+        model (:code:`Pipeline`):
+            Instance of the Transformers pipeline to be saved.
+        signatures (:code: `Dict[str, bool | BatchDimType | AnyType | tuple[AnyType]]`)
+            Methods to expose for running inference on the target model. Signatures are
+             used for creating Runner instances when serving model with bentoml.Service
         labels (:code:`Dict[str, str]`, `optional`, default to :code:`None`):
             user-defined labels for managing models, e.g. team=nlp, stage=dev
-        custom_objects (:code:`Dict[str, Any]]`, `optional`, default to :code:`None`):
-            user-defined additional python objects to be saved alongside the model,
-            e.g. a tokenizer instance, preprocessor function, model configuration json
         metadata (:code:`Dict[str, Any]`, `optional`,  default to :code:`None`):
             Custom metadata for given model.
-        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
-            BentoML modelstore, provided by DI Container.
+
     Returns:
-        :obj:`~bentoml.Tag`: A :obj:`tag` with a format `name:version` where `name` is the user-defined model's name, and a generated `version` by BentoML.
+        :obj:`~bentoml.Tag`: A :obj:`tag` with a format `name:version` where `name` is
+        the user-defined model's name, and a generated `version`.
 
     Examples:
 
     .. code-block:: python
 
-        from transformers import AutoModelForQuestionAnswering, AutoTokenizer
         import bentoml
-        model = AutoModelForQuestionAnswering.from_pretrained("gpt2", from_flax=True)
-        tokenizer = AutoTokenizer.from_pretrained("gpt2", from_flax=True)
 
-        # transfer training and modification goes here
-        ...
+        from transformers import pipeline
 
-        tag = bentoml.transformers.save("flax_gpt2", model=model, tokenizer=tokenizer)
-    """
-    if not check_model_type(obj):
-        if not LazyType["ext.TransformersPipeline"](
-            "transformers.pipelines.base.Pipeline"
-        ).isinstance(obj):
-            raise BentoMLException(
-                "`model` is neither a Transformers pipeline nor a Transformers model."
-            )
+        generator = pipeline(task="text-generation")
+        tag = bentoml.transformers.save_model("text-generation-pipeline", generator)
+
+        # load the model back:
+        loaded = bentoml.transformers.load_model("text-generation-pipeline:latest")
+        # or:
+        loaded = bentoml.transformers.load_model(tag)
+    """  # noqa
+    if not LazyType["ext.TransformersPipeline"](
+        "transformers.pipelines.base.Pipeline"
+    ).isinstance(pipeline):
```

Comment:
when you remove everything under `external_typing.transformers`
```suggestion
    if not LazyType["transformers.pipelines.base.Pipeline"](
        "transformers.pipelines.base.Pipeline"
    ).isinstance(pipeline):
```

Reply:
Thanks for the suggestion. Updated to the following.

Transformers types provided other than Pipeline are not supported in BentoML. 
To save models, tokenizers, configs, feature extractors, construct a pipeline 
with the model, tokenizer, config, or feature extractor specified as arguments, 
then call save_model with the pipeline.

```python
import bentoml
from transformers import pipeline, AutoTokenizer, AutoModelForCausalLM

tokenizer = AutoTokenizer.from_pretrained("distilgpt2")
model = AutoModelForCausalLM.from_pretrained("distilgpt2")
generator = pipeline(task="text-generation", model=model, tokenizer=tokenizer)

bentoml.transformers.save_model("text-generation-pipeline", generator)
```

Refers to https://huggingface.co/docs/transformers/main_classes/pipelines for 
more information on pipelines.